### PR TITLE
feat: add `can store rm` & `can upload rm` commands

### PR DESCRIPTION
**Pull request by olizilla**
adds impl, tests, and docs for the `can store rm` and `can upload rm` commands.

```shell
# remove root cid from upload list
$ w3 can upload rm <root-cid>

# remove car from store
$ w3 can store rm <car-cid>
```

License: MIT

### CODE REVIEW

File: lib.js [hunk 1]
```diff
@@ -168,3 +171,32 @@ export function storeListResponseToString (res, opts = {}) {
...
+
+/**
+ * Return validated CARLink or undefined
+ * @param {AnyLink} cid
+ **/
+export function asCarLink (cid) {
+  if (cid.version === 1 && cid.code === CAR.codec.code) {
+    return /** @type {CARLink} */ (cid)
+  }
+}
```
Comment (alanshaw) [inline lib.js:183]: Alternatively you could make this a type assertion function:

```js
/**
 * Return true if CID is a CAR CID and false if not.
 * @param {import('multiformats').UnknownLink} cid
 * @returns {cid is import('multiformats').Link<Uint8Array, typeof CAR.codec.code>}
 */
export function isCarLink (cid) {
  return cid.version === 1 && cid.code === CAR.codec.code
}
```

...and then typescript knows it's a CAR CID later in the code and you don't have to re-assign e.g.:

Screenshot 2023-10-02 at 16 06 14

File: lib.js [hunk 2]
```diff
@@ -168,3 +171,32 @@ export function storeListResponseToString (res, opts = {}) {
...
+ **/
+export function parseCarLink (cidStr, { error } = console) {
+  try {
+    const cid = CID.parse(cidStr.trim())
+    const car = asCarLink(cid)
+    if (car === undefined) {
+      error(`Error: ${cidStr} is not a CAR CID`)
+    }
+    return car
+  } catch (err) {
```
Comment (alanshaw) [inline lib.js:199]:
```suggestion
  } catch {
```

File: lib.js [hunk 3]
```diff
@@ -168,3 +171,32 @@ export function storeListResponseToString (res, opts = {}) {
...
+ * @param {string} cidStr
+ * @param {object} console
+ * @param {(msg: string) => void} console.error
+ **/
+export function parseCarLink (cidStr, { error } = console) {
+  try {
+    const cid = CID.parse(cidStr.trim())
+    const car = asCarLink(cid)
+    if (car === undefined) {
+      error(`Error: ${cidStr} is not a CAR CID`)
```
Comment (alanshaw) [inline lib.js:196]: Kinda weird to destructure `error` from console...

I'd probably just have it throw when not a CAR CID and then in the calling code:

```js
let cid
try {
  cid = parseCarLink(cidStr)
} catch {
  console.error(`Error: invalid CAR CID: ${cidStr}`)
  process.exit(1)
}
```

**Review by alanshaw — APPROVED** [review]
LGTM - a few suggestions but they are non blocking 🙂

File: lib.js (hunk 3 above)
Comment (olizilla) [reply lib.js]: yeah it used to be `parseCarLinkOrExit` but that was weird too. Have simplified it again.

```js
export function parseCarLink (cidStr) {
  try {
    return asCarLink(CID.parse(cidStr.trim()))
  } catch {}
}
```

calling code

```js
const shard = parseCarLink(cidStr)
if (!shard) {
  console.error(`Error: ${cidStr} is not a CAR CID`)
  process.exit(1)
}
```

**Review by travis — APPROVED** [review]

**Pull request merged**